keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,947 @@
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- (pkg/kubelet/eviction): Sep 11th 2023 - 69.9
+- (pkg/kubelet/stats): Sep 11th 2023 - 77.9
+- (pkg/kubelet/server/stat): Sep 11th 2023 - 55
+
+This KEP will need to cover unit tests where dedicatedImageFs is true.  
+There is currently little test coverage when a separate imagefs is used.  [Issue-120061](https://github.com/kubernetes/kubernetes/issues/120061) has been created to help resolve this.
+
+##### Integration tests
+
+Kubelet is difficult to test via integration.  Typically these type of tests are done with e2e tests.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+This code affects stats, eviction and the summary API.  
+
+There should be e2e tests for each of these components with a split disk.  
+However, there are a few complications with this goal.
+
+1. E2E tests around eviction with a single disk are currently [containerd-eviction](https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-eviction) and [crio-eviction](https://testgrid.k8s.io/sig-node-containerd#node-kubelet-containerd-eviction) failing.  
+
+2. There is zero test coverage around a separate image filesystem.  There is an issue to improve this at the unit test level.
+
+1 can be addressed by investigating the eviction tests and figure out the root cause of these failures.
+
+As part of this KEP, we should add testing around separate disks in upstream Kubernetes.  Since this is already a supported use case in Kubelet, there should be testing around this.
+
+Kubelet/CRIO should be set up with configuration for a separate disk.  Eviction and Summary E2E tests should be added in the case of a separate disk.
+
+And tests for split image filesystem should be added.
+
+E2E Test Use Cases addition:
+
+- E2E tests for summary api with separate disk.
+  - Separate Disk - ImageFs reports separate disk from root when disk is mounted.
+  - Split Disk - Writeable layer on Node, read-only layer on ImageFs
+- E2E tests for eviction api with separate disk
+  - Replicate existing eviction e2e tests with separate disk and split disk.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled.
+- Support for split filesystem.
+- CRI API merged.
+- Support for CRI-O split disk.
+- Image Disk E2E Test Support
+
+#### Beta
+
+- Containerd support
```

Comment:
this continues the conversation in  https://github.com/kubernetes/enhancements/pull/4198/files#r1324961590 but I'm dubious as to whether we actually need this

Reply:
I changed this around.  Figured I put some notes on Alpha to Beta promotion and I'll leave the section empty for beta currently.  We can always update this as we have more information.